### ISSUE TITLE
What is the specific operation process of the tf.stop.gradient？

### ISSUE THREAD

**Issue by lemonltea** [tensorflow/tensorflow#49001]
What is the specific implementation process of the tf.stop.gradient？

My code is:

def round_through(x):
    '''Element-wise rounding to the closest integer with full gradient propagation.
    A trick from [Sergey Ioffe](http://stackoverflow.com/a/36480182)
    a op that behave as f(x) in forward mode,
    but as g(x) in the backward mode.
    '''
    rounded = tf.round(x)
    return x + tf.stop_gradient(rounded-x)

I want to know the specific operation process of stop gradient and the operational formula of stop gradient.

Thank you very much.

**Comment by saikumarchalla** [comment]
@lemonltea Kindly open a stackoverflow issue for this as it is not a bug or feature request, Please post this kind of support questions at Stackoverflow. There is a big community to support and learn from your questions.
Thanks!

**Comment by saikumarchalla** [comment]
@lemonltea tf.stop_gradient() is an operation that acts as the identity function in the forward direction, but stops the accumulated gradient from flowing through that operator in the backward direction. It does not prevent backpropagation altogether, but instead prevents an individual tensor from contributing to the gradients that are computed for an expression. The documentation for the [operation](https://www.tensorflow.org/api_docs/python/tf/stop_gradient) has more details about the operation, and when to use it.

In short, tf.stop_gradient provides a way to not compute gradient with respect to some variables during back-propagation.

Please check this SO [link](https://stackoverflow.com/questions/33727935/how-to-use-stop-gradient-in-tensorflow) where they have explained in a very detailed manner. Thanks!

**Comment by lemonltea** [comment]
> @lemonltea tf.stop_gradient() is an operation that acts as the identity function in the forward direction, but stops the accumulated gradient from flowing through that operator in the backward direction. It does not prevent backpropagation altogether, but instead prevents an individual tensor from contributing to the gradients that are computed for an expression. The documentation for the [operation](https://www.tensorflow.org/api_docs/python/tf/stop_gradient) has more details about the operation, and when to use it.
> 
> In short, tf.stop_gradient provides a way to not compute gradient with respect to some variables during back-propagation.
> 
> Please check this SO [link](https://stackoverflow.com/questions/33727935/how-to-use-stop-gradient-in-tensorflow) where they have explained in a very detailed manner. Thanks!

THANKS
